Library/Homebrew/cmd/update.rb

```diff
@@ -45,13 +47,59 @@ def update
 
     # automatically tap any migrated formulae's new tap
     report.select_formula(:D).each do |f|
-      next unless (HOMEBREW_CELLAR/f).exist?
+      next unless (dir = HOMEBREW_CELLAR/f).exist?
       migration = TAP_MIGRATIONS[f]
       next unless migration
       tap_user, tap_repo = migration.split "/"
       install_tap tap_user, tap_repo
+      # update tap for each Tab
+      tabs = dir.subdirs.each.map { |d| Tab.for_keg(Keg.new(d)) }
+      tabs.each { |tab| tab.source["tap"] = "#{tap_user}/homebrew-#{tap_repo}" }
+      tabs.each(&:write)
     end if load_tap_migrations
 
+    # Migrate installed renamed formulae from main Homebrew repository.
+    if load_formula_renames
+      report.select_formula(:D).each do |f|
+        rename = FORMULA_RENAMES[f]
+        next unless rename
+
+        # To check we have renamed formula installed we do:
+        # 1. Check if not empty directory with formula's name exists.
+        # 2. Find symlinked keg for this formula. If no found use first.
+        # 3. Skip if installed formula is not from the main repository.
+        next unless (dir = HOMEBREW_CELLAR/f).directory? && dir.children.length > 0
+        kegs = dir.subdirs.map { |d| Keg.new(d) }
+        keg = kegs.detect { |k| k.linked? || k.optlinked? } || kegs.first
+        tab = Tab.for_keg(keg) if keg
+        next unless tab && ((tap = tab.tap).nil? || tap == "mxcl/master" \
+                            || tap  == "Homebrew/homebrew")
+        next if tab.source["path"] != Formulary.core_path(f).to_s
```

Comment:
OK, that makes sense. I think another reason explaining this block is that it had been implemented before the behaviour of `Formulary` was changed.